### PR TITLE
CLOUDSTACK-10040 Upload volume fails when management server can not r…

### PR DESCRIPTION
**Pull request by yvsubhash**
Problem
--------
Upload volume from internet fails when the management server cannot reach the URL ( internet)
During the upload volume, we check the validity of the URL through management server first and then download the volume to secondary storage through SSVM.
We have customer environments where the management server doesn’t have internet access ( possibility in some customer environments as per their security policies). This will now cause a failure even though the SSVM has internet connection

Solution
---------
introduced a new global configuration parameter validate.url.existence which if true would attempt the validation of provided URL, else the validation would be skipped

### CODE REVIEW

**Comment by SowjanyaPatha** [comment]
LGTM for Testing.
Initial block the URL in MS from which you want to download the volume.
Set the global parameter "validate.url.existence" to true.
Try to upload volume it fails.
Set the validate.url.existence parameter in Global Settings to false and restart the management server.
Now the volumes gets successfully uploaded.

**Comment by rhtyd** [comment]
LGTM, did not test it.
@blueorangutan package

File: server/src/com/cloud/storage/VolumeApiServiceImpl.java [hunk 1]
```diff
@@ -404,7 +416,10 @@ private boolean validateVolume(Account caller, long ownerId, Long zoneId, String
             }
             UriUtils.validateUrl(format, url);
         // check URL existence
```
Comment (rhtyd) [inline server/src/com/cloud/storage/VolumeApiServiceImpl.java]: @yvsubhash can you fix the comment indentation as well, thanks

**Comment by rhtyd** [comment]
@blueorangutan test

File: server/src/com/cloud/storage/VolumeApiServiceImpl.java (hunk 1 above)
Comment (yvsubhash) [reply server/src/com/cloud/storage/VolumeApiServiceImpl.java]: @rhtyd  its taken care.

File: server/src/com/cloud/storage/VolumeApiServiceImpl.java [hunk 2]
```diff
@@ -403,8 +415,11 @@ private boolean validateVolume(Account caller, long ownerId, Long zoneId, String
                 throw new InvalidParameterValueException("File:// type urls are currently unsupported");
             }
             UriUtils.validateUrl(format, url);
-        // check URL existence
-        UriUtils.checkUrlExistence(url);
+
+            // check URL existence
+            if(ValidateURLExistence.value()) {
```
Comment (rhtyd) [inline server/src/com/cloud/storage/VolumeApiServiceImpl.java]: I see some similar changes in https://github.com/apache/cloudstack/pull/2074 as well

**Comment by rhtyd** [comment]
LGTM, @DaanHoogland can you comment on the changes since they appear similar to one of your PRs /cc @nvazquez @borisstoyanov

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1184
 Hypervisor xenserver
 NetworkType Advanced
 Passed=114
 Failed=7
 Skipped=40

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_router_dnsservice.py

 * test_router_dns_guestipquery Failed

* test_scale_vm.py

 * ContextSuite context=TestScaleVm>:setup Failing since 44 runs

* test_volumes.py

 * test_06_download_detached_volume Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 45 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 172 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 167 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 167 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_disable_oobm_ha_state_ineligible
test_hostha_configure_default_driver
test_hostha_enable_ha_when_host_disabled
test_hostha_enable_ha_when_host_disconected
test_hostha_enable_ha_when_host_in_maintenance
test_hostha_kvm_host_degraded
test_hostha_kvm_host_fencing
test_hostha_kvm_host_recovering
test_remove_ha_provider_not_possible
test_configure_ha_provider_invalid
test_configure_ha_provider_valid
test_ha_configure_enabledisable_across_clusterzones
test_ha_disable_feature_invalid
test_ha_enable_feature_invalid
test_ha_list_providers
test_ha_multiple_mgmt_server_ownership
test_ha_verify_fsm_available
test_ha_verify_fsm_degraded
test_ha_verify_fsm_fenced
test_ha_verify_fsm_recovering
test_hostha_configure_default_driver
test_hostha_configure_invalid_provider
test_hostha_disable_feature_valid
test_hostha_enable_feature_valid
test_hostha_enable_feature_without_setting_provider
test_list_ha_for_host
test_list_ha_for_host_invalid
test_list_ha_for_host_valid
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_outofbandmanagement_nestedplugin.py
test_portforwardingrules.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by yvsubhash** [comment]
@rhtyd  Can we assume this is merge ready as there are no comments from @DaanHoogland  and others

**Comment by rhtyd** [comment]
@yvsubhash yes, let's wait by end of week and merge this if @DaanHoogland @borisstoyanov don't reply.
@DaanHoogland -- please see these changes, they are similar to your PR.

**Comment by yvsubhash** [comment]
@rhtyd  can you please merge this, as  there is no update from Daan

**Comment by rhtyd** [comment]
@DaanHoogland do you have any comments?
@yvsubhash if Daan does not get back to us in next 3 days, we'll proceed with merging this, thanks.

**Comment by DaanHoogland** [comment]
seen this seems to do the same thing. give me some time. I am not biased on which solution to use but want to make sure the same thing is covered

**Comment by DaanHoogland** [comment]
@yvsubhash sorry to say I do not get github notifications anymore. I will that. I have done the exact same thing as you but as part of a slightly bigger refactor and for that reason I would appreciate it if we merge mine.
/cc @rohit @wido

**Comment by yvsubhash** [comment]
@DaanHoogland  merging your pr is fine. But I would like keep this open till #2074 is merged

**Comment by rhtyd** [comment]
Thanks for the pr @yvsubhash, closing this as 2074 has been merged.

**Pull request closed without merging**